naga/src/back/spv/block.rs

```diff
@@ -2034,14 +2034,33 @@ impl<'w> BlockContext<'w> {
         }
     }
 
+    /// Generate one or more SPIR-V blocks for `naga_block`.
+    ///
+    /// Use `label_id` as the label for the SPIR-V entry point block.
+    ///
+    /// If control reaches the end of the SPIR-V block, terminate it according
+    /// to `exit`.
+    ///
+    /// If the block contains [`Break`] or [`Continue`] statements,
+    /// `loop_context` supplies the labels of the SPIR-V blocks to jump to. If
+    /// either of these labels are `None`, then it should have been a Naga
+    /// validation error for the corresponding statement to occur in this
+    /// context.
+    ///
+    /// Return `true` if control can reach the end of the block. This means that
+    /// `exit` was used. Otherwise, return `false`, indicating that `exit` was
+    /// unused.
```

Comment:
nitpick(non-blocking): It initially caused me some confusion that the return type of this function was only consumed by itself recursively. Maybe a start-of-recursion wrapper around this that discards the return type (and making this function itself private, renaming to indicate it's a helper) would be interesting?

Reply:
I'll implement the suggestion above, and then let's look this over.

As I remarked in an earlier comment, I think the other call sites might want to be more explicit about why they're ignoring the return value.

Reply:
Okay, I've done this, in a prep refactoring commit. It let us make two other types private, and reduce the number of arguments to the public function, so this was a win.